scripts/update-command.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: MIT
+
+"""
+A Python script to update the common contents of a command across all languages.
+
+Usage:
+    python3 scripts/update-command.py [-c] [-u] [-n] <PLATFORM> <FILENAME> [COMMAND]
+
+Options:
+    -c, --common-part COMMON_PART
+        Specify the common part to be modified (any content between double brackets will be ignored).
+    -u, --updated-common-part UPDATED_COMMON_PART
+        Specify the updated common part (any content between double brackets will be ignored).
+    -n, --dry-run
+        Show what changes would be made without actually modifying the page.
+
+
+Examples:
+    1. Update 'cargo' page interactively:
+       python3 scripts/update-command.py common cargo
+
+    2. Show what changes would be made by updating `sudo apt install {{}}` in 'apt' page to `sudo apt install {{}} --no-confirm`:
+       python3 scripts/update-command.py --dry-run -c "sudo apt install {{}}" -u "sudo apt install {{}} --no-confirm" linux apt
+"""
+
+from pathlib import Path
+import os
+import re
+import argparse
+import sys
+from functools import reduce
+import logging
+
+
+# Backporting Python 3.9 feature
+class str(str):
+    def removesuffix(self, suffix: str) -> str:
+        return self[: -len(suffix)] if suffix and self.endswith(suffix) else self
+
+
+class MyFormatter(logging.Formatter):
+    grey = "\x1b[0;30m"
+    yellow = "\x1b[33;20m"
+    red = "\x1b[31;20m"
+    bold_red = "\x1b[31;1m"
+    reset = "\x1b[0m"
+    format = "%(levelname)s: %(message)s (%(filename)s:%(lineno)d)"
+
+    FORMATS = {
+        logging.INFO: grey + format + reset,
+        logging.WARNING: yellow + format + reset,
+        logging.ERROR: red + format + reset,
+    }
+
+    def format(self, record):
+        log_fmt = self.FORMATS.get(record.levelno)
+        formatter = logging.Formatter(log_fmt)
+        return formatter.format(record)
+
+
+logger = logging.getLogger(__name__)
+logger.propagate = False
+
+ch = logging.StreamHandler()
+ch.setFormatter(MyFormatter())
+
+logger.addHandler(ch)
+
+
+def get_locales(base_path: Path) -> list[str]:
+    return [
+        d.name.split(".")[1]
+        for d in base_path.iterdir()
+        if d.is_dir() and d.name.startswith("pages.")
+    ]
+
+
+def get_tldr_pages(platform_path: Path) -> list[Path]:
+    return [f for f in platform_path.iterdir() if f.is_file() and f.suffix == ".md"]
+
+
+def take_command_with_common_part(commands: list[str], common_part: str) -> str:
+    return next(
+        (
+            f"`{command}`"
+            for command in commands
+            if remove_placeholders(command) == common_part
+        ),
+        None,
+    )
+
+
+def get_commands_of_page(page_text: str) -> list[str]:
+    command_pattern = re.compile(r"`([^`]+)`")
+    return re.findall(command_pattern, page_text)
+
+
+def find_command_with_common_part(common_part: str, page_text: str) -> list[str]:
+    commands = get_commands_of_page(page_text)
```

Comment:
I think it's more precise to name the variable `commands` as `command_examples`.